src/EditorFeatures/CSharp/EventHookup/EventHookupSessionManager_EventHookupSession.cs

```diff
@@ -257,19 +257,15 @@ private string GetNameObjectPart(IEventSymbol eventSymbol, SyntaxToken plusEqual
                     // This is expected -- it means the last thing is(probably) the event name. We 
                     // already have that in eventSymbol. What we need is the LHS of that dot.
 
-                    var lhs = memberAccessExpression.Expression;
+                    var lhs = memberAccessExpression.Expression.GetRightmostName();
 
-                    if (lhs is MemberAccessExpressionSyntax lhsMemberAccessExpression)
+                    if (lhs is GenericNameSyntax lhsGenericNameSyntax)
                     {
-                        // Okay, cool.  The name we're after is in the RHS of this dot.
-                        return lhsMemberAccessExpression.Name.ToString();
+                        // For generic we must exclude type variables
+                        return lhsGenericNameSyntax.Identifier.Text;
                     }
 
-                    if (lhs is NameSyntax lhsNameSyntax)
-                    {
-                        // Even easier -- the LHS of the dot is the name itself
-                        return lhsNameSyntax.ToString();
-                    }
+                    return lhs.ToString();
```

Comment:
GetRightmostName could return null (at least it's annotated as such), does this need a null check? My gut is it seems like you'd need to have something malformed to the left of the period but still somehow have succeeded in binding the event, so that seems like it'd be a difficult thing to achieve.

Reply:
@jasonmalinowski Nice catch!

But will it really be something malformed? I think it can be null in a completely valid scenario like this?

```
using System;

class C
{
    public event EventHandler MyEvent;
    
    public static C CreateC()
    {
        return new C();
    }
    
    public void M2()
    {
        CreateC().MyEvent +=
    }
}
```

Here `memberAccessExpression.Expression` is InvocationExpressionSyntax where GetRightmostName returns null. We need to fall into `syntaxFactsService` below when it's null, I think.

Reply:
@Youssef1313 Ah right, of course. I was mentally reading through the GetRightmostName() then forgetting that we're passing in the LHS of the original dot, so yeah. If we don't have a test for that case, we should add one regardless,